src/main/kotlin/ch/hevs/cloudio2/cloud/services/AuthenticationService.kt

```diff
@@ -113,23 +135,36 @@ class AuthenticationService(var userRepository: UserRepository, var endpointCred
                 "check_topic" -> {
                     val permission = Permission.valueOf((message.messageProperties.headers["permission"] as String).toUpperCase())
                     val routingKey = (message.messageProperties.headers["routing_key"] as String).split(".")
-                    val password = message.messageProperties.headers["password"]?.toString()
+                    val UUIDpattern = "\\b[0-9a-f]{8}\\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\\b[0-9a-f]{12}\\b".toRegex()
 
                     if(routingKey.size<2)
                        return "deny"
 
-                    val endpointPermission =  when(password) {
-                        null -> {
-                            endpointCredentialRepository.findById(id).get().permissions.getOrDefault(routingKey[1], Permission.DENY)
+                    when(UUIDpattern.matches(id)) {
```

Comment:
If we do not check the presence of a password here, a user could be created that can access an endpoint without having the rights to.

Reply:
When the action is "check_topic", we only get null as password. The rabbitListener only return the password when the action is "login"